diff --git a/event_log_track_auth/event_log_track_auth.module b/event_log_track_auth/event_log_track_auth.module
index cbd263d..7f8e542 100755
--- a/event_log_track_auth/event_log_track_auth.module
+++ b/event_log_track_auth/event_log_track_auth.module
@@ -44,13 +44,19 @@ function event_log_track_auth_form_submit($form, $form_state, $form_id) {
       break;
 
     case 'user_pass':
+      $uid = 0;
+      $account = $form_state->getValue('account');
+      if (isset($account)) {
+        $uid = $account->id();
+      }
+
       $log = array(
         'operation' => 'request password',
         'description' => t('%user (uid %uid)', array(
           '%user' => $form_state->getValue('name'),
-          '%uid' => $form_state->getValue('account')->id(),
+          '%uid' => $uid,
         )),
-        'ref_numeric' => $form_state->getValue('account')->id(),
+        'ref_numeric' => $uid,
         'ref_char' => $form_state->getValue('name'),
       );
       break;
